### PR TITLE
Update secp256k1-zkp

### PR DESCRIPTION
**Pull request by onvej-sl**
### TLDR

This PR updates the `secp256k1-zkp` submodule and fixes https://github.com/trezor/trezor-firmware/issues/2085. 

### Details
* The commit https://github.com/trezor/trezor-firmware/commit/b770c5cb91322336599eb1979bf939a0c677eb1e updates the `secp256k1-zkp` submodule. Some configuration options were removed and the way the precomputed tables are generated changed:
  * The configuration options `USE_NUM_NONE`, `USE_FIELD_INV_BUILTIN` and `USE_SCALAR_INV_BUILTIN` were removed in https://github.com/ElementsProject/secp256k1-zkp/commit/1f233b3fa05eb29a744487e0682d925055fb0d4c. Since then the default behaviour is as these options were set.
  * The options `DUSE_FIELD_10X26` and `DUSE_SCALAR_8X32` enforce the use of the 32 x 32 bit multiplication. These options were replaced with the option `SECP256K1_WIDEMUL_INT64` in https://github.com/ElementsProject/secp256k1-zkp/commit/79f1f7a4f123765cf07be92ae894d882c5845191. This option is autmatically set if `__SIZEOF_INT128__` is not defined, which is our case, and that implies the 32 x 32 bit multiplication is used.
  * The option `USE_ECMULT_STATIC_PRECOMPUTATION` was removed in https://github.com/ElementsProject/secp256k1-zkp/commit/22dc2c0a0dc3b321e72253f492cfa8bcbf00169b. Since then the default behaviour is as this option was set, that is some tables  are precomputed in compile time (instead of in run time). Since https://github.com/ElementsProject/secp256k1-zkp/commit/22dc2c0a0dc3b321e72253f492cfa8bcbf00169b the tables are included in the repository, that is there is no need to precompute them in compile-time.
* The commits https://github.com/trezor/trezor-firmware/commit/e0c7dca17bde623dc4d61a4c4860d013c5de7497 and https://github.com/trezor/trezor-firmware/commit/a1ac7ef2f6d5500f0d37410ea8702445a81edebf fixes https://github.com/trezor/trezor-firmware/issues/2085
* The commit https://github.com/trezor/trezor-firmware/commit/0a4abcb507a37b15c5ab0bac5e5c1073d2c16f63 removes code that is no more needed after the update, see [this](https://github.com/ElementsProject/secp256k1-zkp/blob/master/include/secp256k1_schnorrsig.h#L113-L117).

### CODE REVIEW

**Review by prusnak — APPROVED** [review]
untested ACK

**Comment by matejcik** [comment]
> * The option `USE_ECMULT_STATIC_PRECOMPUTATION` was removed

does this mean that the binary size is ~same as before?
i.e., did we have the precomputed tables before too, except now they are separate obj files?

**Comment by onvej-sl** [comment]
> does this mean that the binary size is ~same as before?

I suppose so but I didn't test it.

> i.e., did we have the precomputed tables before too, except now they are separate obj files?

Yes, this is exactly what happened.

**Comment by onvej-sl** [comment]
> I suppose so but I didn't test it.

This PR doesn't seem to change `code_length` in TT firmware header, which is a bit suspicious.

**Review by andrewkozlik — APPROVED** [review]
LGTM. We can probably remove the reference to gen_context.o from core/embed/rust/build.rs.

File: crypto/Makefile [hunk 1]
```diff
@@ -165,7 +161,7 @@ tests/test_openssl: tests/test_openssl.o $(OBJS)
 	$(CC) $(CFLAGS) tests/test_openssl.o $(OBJS) $(TESTSSLLIBS) -o tests/test_openssl
 
 tests/libtrezor-crypto.so: $(SRCS) secp256k1-zkp.o
```
[the remark this answers is not in the document]
Comment (onvej-sl) [reply crypto/Makefile]: Fixed in https://github.com/trezor/trezor-firmware/pull/2261/commits/711aa04ffd580037405e9ce4b6d606877f2e3f8c.

**Comment by onvej-sl** [comment]
> We can probably remove the reference to gen_context.o from core/embed/rust/build.rs.

Yes, that is true. Done in https://github.com/trezor/trezor-firmware/pull/2261/commits/d24d4a4e5aa50a12d2ef618d6d718fd927d34576.

**Review by andrewkozlik — APPROVED** [review]
